### PR TITLE
Update to Dart 2.13

### PR DESCRIPTION
**Pull request by todbachman-wf**
### Problem:
We're updating the Dart ecosystem at workiva to use Dart 2.13 everywhere. This repo still uses Dart 2.13 in CI. In addition, when trying to [run this tool recently in a skynet config using a Dart 2.13.4 base image](https://wf-skynet-hrd.appspot.com/apps/test/smithy/3059379/1/graph_app/dart-lsif#summary) I encountered the following error when the lsif_indexer is precompiled:

```base
2021-09-02T19:05:33.857664103Z Precompiling executables... 
2021-09-02T19:05:33.857693521Z Precompiled lsif_indexer:compare.
2021-09-02T19:05:38.143369551Z Failed to precompile lsif_indexer:lsif_indexer:
2021-09-02T19:05:38.143394774Z /root/.pub-cache/hosted/pub.dartlang.org/analyzer-0.41.1/lib/src/error/best_practices_verifier.dart:258:50: Error: The property 'displayString' is defined in multiple extensions for 'TargetKind' and neither is more specific.
2021-09-02T19:05:38.143399154Z  - 'TargetKind' is from 'package:meta/meta_meta.dart' ('/root/.pub-cache/hosted/pub.dartlang.org/meta-1.7.0/lib/meta_meta.dart').
2021-09-02T19:05:38.143410040Z Try using an explicit extension application of the wanted extension or hiding unwanted extensions from scope.
2021-09-02T19:05:38.143414352Z         var kindNames = kinds.map((kind) => kind.displayString).toList()
2021-09-02T19:05:38.143417961Z                                                  ^^^^^^^^^^^^^
2021-09-02T19:05:38.143420561Z /root/.pub-cache/hosted/pub.dartlang.org/analyzer-0.41.1/lib/src/error/best_practices_verifier.dart:1950:14: Context: This is one of the extension members.
2021-09-02T19:05:38.143424830Z   String get displayString {
2021-09-02T19:05:38.143427197Z              ^^^^^^^^^^^^^
2021-09-02T19:05:38.143429543Z /root/.pub-cache/hosted/pub.dartlang.org/meta-1.7.0/lib/meta_meta.dart:91:14: Context: This is one of the extension members.
2021-09-02T19:05:38.143460458Z   String get displayString {
2021-09-02T19:05:38.143478517Z              ^^^^^^^^^^^^^
2021-09-02T19:05:38.143504212Z /root/.pub-cache/hosted/pub.dartlang.org/analyzer-0.41.1/lib/src/error/best_practices_verifier.dart:260:36: Error: The getter 'commaSeparatedWithOr' isn't defined for the class 'List<dynamic>'.
2021-09-02T19:05:38.143515697Z  - 'List' is from 'dart:core'.
2021-09-02T19:05:38.143527934Z Try correcting the name to the name of an existing getter, or defining a getter or field named 'commaSeparatedWithOr'.
2021-09-02T19:05:38.143530923Z         var validKinds = kindNames.commaSeparatedWithOr;
2021-09-02T19:05:38.143546905Z                                    ^^^^^^^^^^^^^^^^^^^^
2021-09-02T19:05:38.173421461Z  $ echo 'LSIF Indexer activation failed'
2021-09-02T19:05:38.173434435Z LSIF Indexer activation failed
2021-09-02T19:05:38.173437178Z  $ exit 0
2021-09-02T19:05:38.173617766Z  $ pub global run lsif_indexer -o dump.lsif
```

According to the [Changelog for the meta package](https://pub.dev/packages/meta/changelog), a change was made in the 1.5.0 release that is incompatible with some versions of analyzer, the one this package is using appears to be one of them.

### Solution:
- Updated the `Dockerfile` to use Dart 2.13.4
- Limited the range for `meta` to `>=1.2.2 <1.5.0` to avoid newer versions incompatible with the version of `analyzer` we're using
- Moved CI to a github workflow
- Updated the commands used to run analysis, formatting, and tests in CI to use the new `dart {command}`s.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by evanweible-wf — APPROVED** [review]

File: .github/workflows/dart_ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - id: install
+        name: Install dependencies
+        run: pub get
+
+      - name: Analyze project source
+        run: dart analyze
+        if: always() && steps.install.outcome == 'success'
+
+      - name: Verify formatting
+        run: dart format --set-exit-if-changed .
```
Comment (evanweible-wf) [inline .github/workflows/dart_ci.yml:36]: #nit I'm not sure this makes much of a difference, but this is the equivalent to `--dry-run`:
```suggestion
        run: dart format --output=none --set-exit-if-changed .
```

**Review by evanweible-wf — DISMISSED** [review]

**Review by evanweible-wf — APPROVED** [review]

**Comment by todbachman-wf** [comment]
QA +1. Passing CI should suffice.

**Comment by todbachman-wf** [comment]
@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**